sys/cbor/cbor.c

```diff
@@ -400,7 +400,7 @@ static size_t decode_bytes(const cbor_stream_t *s, size_t offset, char *out, siz
         return 0;
     }
 
-    if (length < bytes_length + 1) {
+    if (bytes_length == SIZE_MAX || length < bytes_length + 1) {
```

Comment:
How does this fix work? Does "decode_int()" return 0xFF....FF, causing "bytes_length +1" to overflow to 0?

Reply:
if `bytes_length` equals `SIZE_MAX` then `bytes_length + 1` overflows to zero making `length < bytes_length + 1` evaluate to false, however `bytes_length` is later on passed to memcpy as `memcpy(out, &s->data[offset + bytes_start], bytes_length)`.

Reply:
He meant to say "yes".